ndmg/scripts/ndmg_dwi_pipeline.py

```diff
@@ -108,61 +108,62 @@ def ndmg_dwi_worker(
     for arg, value in args.items():
         print(f"{arg} = {value}")
 
-    fmt = "_adj.csv"
+    # initial assertions
+    if vox_size not in ["1mm", "2mm"]:
+        raise ValueError("Voxel size not supported. Use 2mm or 1mm")
 
-    # make output directory
+    print("Checking inputs...")
+    for file_ in [t1w, bvals, bvecs, dwi, atlas, mask, *parcellations]:
+        if not os.path.isfile(file_):
+            raise FileNotFoundError(f"Input {file_} not found. Exiting ndmg.")
+        else:
+            print(f"Input {file_} found.")
+
+    # time ndmg execution
     startTime = datetime.now()
-    namer = gen_utils.NameResource(dwi, t1w, atlas, outdir)
-    Path(outdir).mkdir(parents=True, exist_ok=True)
-    paths = {
-        "prep_dwi": "dwi/preproc",
-        "prep_anat": "anat/preproc",
-        "reg_anat": "anat/registered",
-        "fiber": "dwi/fiber",
-        "tensor": "dwi/tensor",
-        "conn": "dwi/roi-connectomes",
-    }
 
+    # initial variables
+    outdir = Path(outdir)
+    dwi_name = gen_utils.get_filename(dwi)
+
+    # make output directory
     print("Adding directory tree...")
-    namer.add_dirs(paths, labels, ["conn"])
+    parcellations = gen_utils.as_list(parcellations)
+    gen_utils.make_initial_directories(outdir, parcellations=parcellations)
 
     # generate list of connectome file locations
-    labels = gen_utils.as_list(labels)
     connectomes = []
-    for label in labels:
-        filename = namer.get_label(label)
-        folder = namer.dirs["output"]["conn"][filename]
-        derivative = f"{namer.__subi__}_{filename}_measure-spatial-ds{fmt}"
-        connectomes.append(namer.name_derivative(folder, derivative))
+    for parc in parcellations:
+        name = gen_utils.get_filename(parc)
+        folder = outdir / f"connectomes/{name}"
+        connectome = f"{dwi_name}_{name}_connectome.csv"
+        connectomes.append(str(folder / connectome))
 
     warm_welcome = welcome_message(connectomes)
     print(warm_welcome)
 
-    if vox_size not in ["1mm", "2mm"]:
-        raise ValueError("Voxel size not supported. Use 2mm or 1mm")
-
     # -------- Preprocessing Steps --------------------------------- #
 
-    # Perform eddy correction
-    dwi_prep = f'{namer.dirs["output"]["prep_dwi"]}/eddy_corrected_data.nii.gz'
-    preproc_dir = namer.dirs["output"]["prep_dwi"]
+    # set up directories
+    prep_dwi: Path = outdir / "dwi/preproc"
+    eddy_corrected_data: Path = prep_dwi / "eddy_corrected_data.nii.gz"
 
     # check that skipping eddy correct is possible
     if skipeddy:
-        # do it anyway if dwi_prep doesnt exist
-        if not os.path.isfile(dwi_prep):
+        # do it anyway if eddy_corrected_data doesnt exist
+        if not eddy_corrected_data.is_file():
             print("Cannot skip preprocessing if it has not already been run!")
             skipeddy = False
```

Comment:
here, `skipeddy` is just toggled, rather than the zillions of checks that were happening before.
didn't do that for `skipreg`, but it would probably be cleaner (was just faster to code it the other way)